File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/basket_margins.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json          = Json(JsonConfiguration.Stable)
// val basketMargins = json.parse(BasketMargins.serializer(), jsonString)

package BasketMargins

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class BasketMargins (
    val data: Data? = null,
    val status: String? = null
)

@Serializable
data class Data (
    val final: Final? = null,
    val initial: Final? = null,
    val orders: List<Final>? = null
)

@Serializable
data class Final (
    val additional: Long? = null,
    val bo: Long? = null,
    val cash: Long? = null,
    val exchange: String? = null,
    val exposure: Double? = null,

    @SerialName("option_premium")
    val optionPremium: Double? = null,

    val pnl: Pnl? = null,
    val span: Double? = null,
    val total: Double? = null,
    val tradingsymbol: String? = null,
    val type: String? = null,

    @SerialName("var")
    val finalVar: Long? = null
)

@Serializable
data class Pnl (
    val realised: Long? = null,
    val unrealised: Long? = null
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/basket_margins.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Debug, Serialize, Deserialize)]
pub struct BasketMargins {
    pub data: Option<Data>,
    pub status: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Data {
    #[serde(rename = "final")]
    pub data_final: Option<Final>,
    pub initial: Option<Final>,
    pub orders: Option<Vec<Final>>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Final {
    pub additional: Option<i64>,
    pub bo: Option<i64>,
    pub cash: Option<i64>,
    pub exchange: Option<String>,
    pub exposure: Option<f64>,
    pub option_premium: Option<f64>,
    pub pnl: Option<Pnl>,
    pub span: Option<f64>,
    pub total: Option<f64>,
    pub tradingsymbol: Option<String>,
    #[serde(rename = "type")]
    pub final_type: Option<String>,
    pub var: Option<i64>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Pnl {
    pub realised: Option<i64>,
    pub unrealised: Option<i64>,
}
